### PR TITLE
fix: Use getenv() instead of _SERVER for vars

### PR DESCRIPTION
**Pull request by darcywong00**
Relates to keymanapp/shared-sites#7

In draft - I still need to run through Docker to see how this goes.

### CODE REVIEW

File: script/desktop/10.0/exception/index.php [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?php
   require_once('../../../../tools/base.inc.php');
-  define('CACERT_PATH', $_SERVER['DOCUMENT_ROOT'].'/tools/cacert.pem');
+  $env = getenv();
+
+  define('CACERT_PATH', $env['DOCUMENT_ROOT'].'/tools/cacert.pem');
```
Comment (mcdurdin) [inline script/desktop/10.0/exception/index.php:5]: DOCUMENT_ROOT is also a standard $_SERVER prop -- see https://www.php.net/manual/en/reserved.variables.server.php for standard props

File: script/desktop/10.0/submitdiag/index.php [hunk 2]
```diff
@@ -1,6 +1,8 @@
 <?php
   require_once('../../../../tools/base.inc.php');
-  if(!isset($_SERVER['argv'])) {
+
+  $env = getenv();
+  if(!isset($env['argv'])) {
```
Comment (mcdurdin) [inline script/desktop/10.0/submitdiag/index.php:5]: standard prop :grin:

File: script/desktop/10.0/submitdiag/index.php [hunk 3]
```diff
@@ -84,9 +86,9 @@
   echo "<result>$topic_id</result>";
 
   function CallDiscourse($path, $data, $method = 'GET', $username = '') {
-    $discourse_key = $_SERVER['api_keyman_com_discourse_key'];
-    $discourse_site = $_SERVER['api_keyman_com_discourse_site'];
-    $discourse_username = $_SERVER['api_keyman_com_discourse_username'];
+    $discourse_key = $env['api_keyman_com_discourse_key'];
+    $discourse_site = $env['api_keyman_com_discourse_site'];
+    $discourse_username = $env['api_keyman_com_discourse_username'];
```
Comment (mcdurdin) [inline script/desktop/10.0/submitdiag/index.php:91]: These need a `global $env;` to be able to read the global. Yeeeuch PHP

```suggestion
    global $env;
    $discourse_key = $env['api_keyman_com_discourse_key'];
    $discourse_site = $env['api_keyman_com_discourse_site'];
    $discourse_username = $env['api_keyman_com_discourse_username'];
```

File: script/increment-download/increment-download.php [hunk 4]
```diff
@@ -11,12 +11,13 @@
...
 
   $mssql = Keyman\Site\com\keyman\api\Tools\DB\DBConnect::Connect();
+  $env = getenv();
 
   header('Link: <' . KeymanHosts::Instance()->api_keyman_com .'/schemas/increment-download/1.0/increment-download.json#>; rel="describedby"');
 
   $AllowGet = isset($_REQUEST['debug']);
 
-  if(!$AllowGet && $_SERVER['REQUEST_METHOD'] != 'POST') {
+  if(!$AllowGet && $env['REQUEST_METHOD'] != 'POST') {
```
Comment (mcdurdin) [inline script/increment-download/increment-download.php:20]: standard prop

File: script/package-version/package-version.php [hunk 5]
```diff
@@ -32,7 +32,8 @@
 
   header('Link: <' . KeymanHosts::Instance()->api_keyman_com . '/schemas/package-version.json#>; rel="describedby"');
 
-  $params = fix_array_params($_SERVER['QUERY_STRING']);
+  $env = getenv();
+  $params = fix_array_params($env['QUERY_STRING']);
```
Comment (mcdurdin) [inline script/package-version/package-version.php:36]: standard prop

**Review by mcdurdin — CHANGES_REQUESTED** [review]
:grin:

File: script/desktop/10.0/exception/index.php (hunk 1 above)
Comment (darcywong00) [reply script/desktop/10.0/exception/index.php:5]:
```suggestion
  $env = getenv();

  define('CACERT_PATH', $_SERVER['DOCUMENT_ROOT'].'/tools/cacert.pem');
```

File: script/desktop/10.0/submitdiag/index.php (hunk 2 above)
Comment (darcywong00) [reply script/desktop/10.0/submitdiag/index.php:5]:
```suggestion
  $env = getenv();
  if(!isset($_SERVER['argv'])) {
```

File: script/increment-download/increment-download.php (hunk 4 above)
Comment (darcywong00) [reply script/increment-download/increment-download.php:20]:
```suggestion
  if(!$AllowGet && $_SERVER['REQUEST_METHOD'] != 'POST') {
```

File: script/package-version/package-version.php (hunk 5 above)
Comment (darcywong00) [reply script/package-version/package-version.php:36]:
```suggestion
  $env = getenv();
  $params = fix_array_params($_SERVER['QUERY_STRING']);
```

Comment (darcywong00) [reply script/package-version/package-version.php]: Reverted in https://github.com/keymanapp/api.keyman.com/pull/213/commits/0ee70c7dd56af88308d3d466b1524eb153e08903

File: script/package-version/package-version.php [hunk 6]
```diff
@@ -32,6 +32,7 @@
 
   header('Link: <' . KeymanHosts::Instance()->api_keyman_com . '/schemas/package-version.json#>; rel="describedby"');
 
+  $env = getenv();
```
Comment (mcdurdin) [inline script/package-version/package-version.php:35]:
```suggestion
```

File: tests/TestUtils.inc.php [hunk 7]
```diff
@@ -16,8 +16,9 @@ static function LoadJSONSchema($filename) {
     }
 
     static function Hostname() {
-      if(isset($_SERVER['api_keyman_com_host'])) {
-        return $_SERVER['api_keyman_com_host'];
+      $env = getenv();
+      if(isset($env['api_keyman_com_host'])) {
+        return $env['api_keyman_com_host'];
```
Comment (mcdurdin) [inline tests/TestUtils.inc.php:21]: I don't know what the cost of `getenv()` is but this could potentially be slow if we call `Hostname()` a lot?

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**